### ISSUE TITLE
`Invalid coercion encountered: ConversionFailure("Float(BigDecimal(\"10.2\"))", "Decimal")`

### ISSUE THREAD

**Issue by ElmutYazid** [prisma/prisma#7580]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v14.8.0            | 
| OS              | darwin             |
| Prisma Client   | 2.24.1             |
| Query Engine    | query-engine 18095475d5ee64536e2f93995e48ad800737a9e4|
| Database        | undefined|

## Query
```
mutation {
  createOnecorresponding_grams(data: {
    calories: 5
    plan: "X"
    size: 5
  }) {
    calories
    plan
    size
  }
}
```

## Logs
```
template%3Dbug_report.md%0A++prisma%3Aclient+%0A++prisma%3Aclient+If+you+want+the+Prisma+team+to+look+into+it%2C+please+open+the+link+above+%F0%9F%99%8F%0A++prisma%3Aclient+To+increase+the+chance+of+success%2C+please+post+your+schema+and+a+snippet+of%0A++prisma%3Aclient+how+you+used+Prisma+Client+in+the+issue.+%0A++prisma%3Aclient+%0A++prisma%3Aclient+++++at+NodeEngine.throwAsyncErrorIfExists+%28%2FUsers%2Fyazid%2Fworkspace%2Felmut_app_v3%2Felmut_backend%2Fnode_modules%2F%40prisma%2Fclient%2Fruntime%2Findex.js%3A28354%3A21%29%0A++prisma%3Aclient+++++at+NodeEngine.handleRequestError+%28%2FUsers%2Fyazid%2Fworkspace%2Felmut_app_v3%2Felmut_backend%2Fnode_modules%2F%40prisma%2Fclient%2Fruntime%2Findex.js%3A27668%3A14%29%0A++prisma%3Aclient+++++at+runMicrotasks+%28%3Canonymous%3E%29%0A++prisma%3Aclient+++++at+processTicksAndRejections+%28internal%2Fprocess%2Ftask_queues.js%3A93%3A5%29%0A++prisma%3Aclient+++++at+NodeEngine.request+%28%2FUsers%2Fyazid%2Fworkspace%2Felmut_app_v3%2Felmut_backend%2Fnode_modules%2F%40prisma%2Fclient%2Fruntime%2Findex.js%3A28288%3A9%29%0A++prisma%3Aclient+++++at+cb+%28%2FUsers%2Fyazid%2Fworkspace%2Felmut_app_v3%2Felmut_backend%2Fnode_modules%2F%40prisma%2Fclient%2Fruntime%2Findex.js%3A35193%3A26%29++%0A++prisma%3Aengine+%7B%0A++prisma%3Aengine+++error%3A+Error%3A+connect+ECONNREFUSED+127.0.0.1%3A57100%0A++prisma%3Aengine+++++++at+TCPConnectWrap.afterConnect+%5Bas+oncomplete%5D+%28net.js%3A1144%3A16%29+%7B%0A++prisma%3Aengine+++++errno%3A+-61%2C%0A++prisma%3Aengine+++++code%3A+%27ECONNREFUSED%27%2C%0A++prisma%3Aengine+++++syscall%3A+%27connect%27%2C%0A++prisma%3Aengine+++++address%3A+%27127.0.0.1%27%2C%0A++prisma%3Aengine+++++port%3A+57100%0A++prisma%3Aengine+++%7D%0A++prisma%3Aengine+%7D++%0A%60%60%60%0A%0A%23%23+Client+Snippet%0A%60%60%60ts%0A%2F%2F+PLEASE+FILL+YOUR+CODE+SNIPPET+HERE%0A%60%60%60%0A%0A%23%23+Schema%0A%60%60%60prisma%0A%2F%2F+PLEASE+ADD+YOUR+SCHEMA+HERE+IF+POSSIBLE%0A%60%60%60%0A&title=PANIC+in+libs%2Fprisma-models%2Fsrc%2Frecord.rs%3A161%3A30%0AInvalid+coercion+encountered%3A+ConversionFailure%28%22Float%28BigDecimal%28%5C%221652%5C%22%29%29%22%2C+%22Decimal%22%29&template=bug_report.md
  prisma:client 
  prisma:client If you want the Prisma team to look into it, please open the link above 🙏
  prisma:client To increase the chance of success, please post your schema and a snippet of
  prisma:client how you used Prisma Client in the issue. 
  prisma:client 
  prisma:client     at NodeEngine.throwAsyncErrorIfExists (/Users/yazid/workspace/elmut_app_v3/elmut_backend/node_modules/@prisma/client/runtime/index.js:28354:21)
  prisma:client     at NodeEngine.handleRequestError (/Users/yazid/workspace/elmut_app_v3/elmut_backend/node_modules/@prisma/client/runtime/index.js:27668:14)
  prisma:client     at NodeEngine.request (/Users/yazid/workspace/elmut_app_v3/elmut_backend/node_modules/@prisma/client/runtime/index.js:28288:20)
  prisma:client     at runMicrotasks (<anonymous>)
  prisma:client     at processTicksAndRejections (internal/process/task_queues.js:93:5)
  prisma:client     at cb (/Users/yazid/workspace/elmut_app_v3/elmut_backend/node_modules/@prisma/client/runtime/index.js:35193:26)  
  prisma:engine { cwd: '/Users/yazid/workspace/elmut_app_v3/elmut_backend/src/prisma' }  
  prisma:engine Search for Query Engine in /Users/yazid/workspace/elmut_app_v3/elmut_backend/node_modules/.prisma/client  
  plusX Execution permissions of /Users/yazid/workspace/elmut_app_v3/elmut_backend/node_modules/.prisma/client/query-engine-darwin are fine  
  prisma:engine { flags: [ '--enable-raw-queries', '--port', '57104' ] }  
  prisma:engine stdout  Starting a postgresql pool with 13 connections.  
  prisma:engine stdou

**Comment by pantharshit00** [comment]
Can you please share the full logs here once? That will help us to point out the part of engines code which is triggering this panic.

Also, if possible, please share your Prisma schema and the client call which triggers this panic.

**Comment by ElmutYazid** [comment]
Yes, here my schema :

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["transactionApi"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model animal {
  animal_id        String           @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  sexe             String           @db.Char(1)
  breed            String           @db.VarChar(80)
  name             String           @db.VarChar(42)
  age              Int              @db.SmallInt
  weight           Decimal          @db.Decimal(5, 2)
  activity_level   String           @db.Char(12)
  body             String           @db.Char(12)
  neutered         Boolean
  previous_food    String           @db.VarChar(42)
  quantity_per_day Int              @db.SmallInt
  calories_per_day Int?             @db.SmallInt
  user_id          String?          @db.Uuid
  created_at       DateTime?        @default(now()) @db.Timestamptz(6)
  elmut_user       elmut_user?      @relation(fields: [user_id], references: [user_id])
  have_pathology   have_pathology[]
}

model delivery_day {
  delivery_day_id String         @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  day             String         @unique @db.VarChar(12)
  subscription    subscription[]
}

model elmut_order {
  order_id               String        @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  amount                 Decimal       @db.Decimal(5, 2)
  portion_size           Int           @db.SmallInt
  frequency              Int           @db.SmallInt
  delivery               DateTime      @db.Timestamptz(6)
  invoice_date           DateTime      @db.Timestamptz(6)
  city                   String        @db.VarChar(80)
  postcode               String        @db.Char(5)
  adress                 String        @db.VarChar(255)
  additional_information String?       @db.VarChar(255)
  created_at             DateTime?     @default(now()) @db.Timestamptz(6)
  subscription_id        String?       @db.Uuid
  beef600                Int           @db.SmallInt
  beef300                Int           @db.SmallInt
  chicken600             Int           @db.SmallInt
  chicken300             Int           @db.SmallInt
  turkey300              Int           @db.SmallInt
  turkey600              Int           @db.SmallInt
  subscription           subscription? @relation(fields: [subscription_id], references: [subscription_id])
}

model elmut_user {
  user_id                String        @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  lastname               String?       @db.VarChar(128)
  firstname              String        @db.VarChar(128)
  email                  String        @unique @db.VarChar(255)
  password               String?       @db.VarChar(255)
  phone                  String?       @db.Char(14)
  entreprise             String?       @db.VarChar(80)
  country                String?       @db.VarChar(64)
  city                   String?       @db.VarChar(128)
  postcode               String?       @db.Char(5)
  address                String?       @db.VarChar(400)
  additional_information String?       @db.VarChar(255)
  user_type              String?       @db.VarChar(32)
  code                   String?       @db.VarChar(64)
  newsletter             Boolean?
  created_at             DateTime?     @default(now()) @db.Timestamptz(6)
  animal                 animal[]
  subscription           subscription?
}

model have_pathology {
  animal_id    String    @db.Uuid
  pathology_id String    @db.Uuid
  animal       animal    @relation(fields: [animal_id], references: [animal_id])
  pathology    pathology @relation(fields: [pathology_id], references: [pathology_id])

  @@id([animal_id, pathology_id])
}

model have_plan {
  subscription_id String       @db.Uuid
  plan_id         String       @db.Uuid
  grams           Int          @db.SmallInt
  plan            plan         @relation(fields: [plan_id], references: [plan_id])
  subscription    subscription @relation(fields: [subscription_id], references: [subscription_id])

  @@id([subscription_id, plan_id])
}

model have_voucher {
  subscription_id String       @db.Uuid
  voucher_id      String       @db.Uuid
  subscription    subscription @relation(fields: [subscription_id], references: [subscription_id])
  voucher         voucher      @relation(fields: [voucher_id], references: [voucher_id])

  @@id([subscription_id, voucher_id])
}

model own_recipe {
  subscription_id String       @db.Uuid
  recipe_id       String       @db.Uuid
  recipe          recipe       @relation(fields: [recipe_id], references: [recipe_id])
  subscription    subscription @relation(fields: [subscription_id], references: [subscription_id])

  @@id([subscription_id, recipe_id])
}

model pathology {
  pathology_id   String           @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  name           String           @unique @db.VarChar(32)
  description    String           @db.VarChar(255)
  have_pathology have_pathology[]
}

model plan {
  plan_id     String      @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  name        String      @unique @db.VarChar(42)
  description String      @db.VarChar(255)
  have_plan   have_plan[]
}

model recipe {
  recipe_id   String       @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  name        String       @unique @db.VarChar(24)
  composition String
  ingredients String
  own_recipe  own_recipe[]
}

model subscription {
  subscription_id           String         @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  portion_size              Int            @db.SmallInt
  second_portion_size       Int            @db.SmallInt
  third_portion_size        Int            @db.SmallInt
  delivery_frequency        Int            @db.SmallInt
  second_delivery_frequency Int            @db.SmallInt
  third_delivery_frequency  Int            @db.SmallInt
  last_delivery             DateTime?      @db.Timestamptz(6)
  delivery                  DateTime?      @db.Timestamptz(6)
  second_delivery           DateTime?      @db.Timestamptz(6)
  third_delivery            DateTime?      @db.Timestamptz(6)
  invoice_date              DateTime?      @db.Timestamptz(6)
  next_invoice_date         DateTime?      @db.Timestamptz(6)
  report_delivery           Int?           @db.SmallInt
  date_report_delivery      DateTime?      @db.Timestamptz(6)
  amount                    Decimal        @db.Decimal(5, 2)
  second_amount             Decimal        @db.Decimal(5, 2)
  third_amount              Decimal        @db.Decimal(5, 2)
  price_per_day             Decimal        @db.Decimal(5, 2)
  delivery_cost             Decimal        @db.Decimal(5, 2)
  package_weight            Int
  cancellation_date         DateTime?      @db.Timestamptz(6)
  cancellation_reason       String?        @db.VarChar(255)
  status                    String         @db.VarChar(24)
  user_id                   String?        @unique @db.Uuid
  delivery_day_id           String?        @db.Uuid
  created_at                DateTime?      @default(now()) @db.Timestamptz(6)
  updated_at                DateTime?      @db.Timestamptz(6)
  stripe_customer_id        String         @db.VarChar(255)
  stripe_subscription_id    String         @db.VarChar(255)
  stripe_price_id           String         @db.VarChar(255)
  stripe_schedule_id        String?        @db.VarChar(255)
  delivery_day              delivery_day?  @relation(fields: [delivery_day_id], references: [delivery_day_id])
  elmut_user                elmut_user?    @relation(fields: [user_id], references: [user_id])
  elmut_order               elmut_order[]
  have_plan                 have_plan[]
  have_voucher              have_voucher[]
  own_recipe                own_recipe[]
}

model voucher {
  voucher_id       String         @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  code             String         @unique @db.VarChar(42)
  percentage       Int            @db.SmallInt
  description      String         @db.VarChar(256)
  state            String         @db.VarChar(18)
  user_application String         @db.VarChar(18)
  start_date       DateTime?      @db.Timestamptz(6)
  end_date         DateTime?      @db.Timestamptz(6)
  have_voucher     have_voucher[]
}

model beef_pricing {
  quantity Int     @id @db.SmallInt
  price    Decimal @db.Decimal(5, 2)
}

model behaviour_coef {
  name String  @id @db.VarChar(32)
  coef Decimal @db.Decimal(5, 2)
}

model breed {
  name     String  @id @db.VarChar(80)
  typology String  @db.VarChar(80)
  coef     Decimal @db.Decimal(5, 2)
}

model chicken_pricing {
  quantity Int     @id @db.SmallInt
  price    Decimal @db.Decimal(5, 2)
}

model conditionning {
  weight Decimal @id @db.Decimal
  size   Decimal @db.Decimal
}

model corresponding_grams {
  calories Decimal @db.Decimal
  plan     String  @db.VarChar(32)
  size     Decimal @db.Decimal

  @@id([calories, plan])
}

model delivery_frequency {
  quantity  Int @id @db.SmallInt
  frequency Int @db.SmallInt
}

model food_transition {
  plan     String @db.VarChar(16)
  quantity Int    @db.SmallInt
  t1       Int    @db.SmallInt
  t2       Int    @db.SmallInt
  t3       Int    @db.SmallInt
  boudin   String @db.VarChar(80)

  @@id([plan, quantity])
}

model morphology {
  name String  @id @db.VarChar(32)
  coef Decimal @db.Decimal(5, 2)
}

model token {
  user_id    String    @id @db.Uuid
  token      String?   @db.VarChar(256)
  created_at DateTime? @db.Timestamptz(6)
}

model typology {
  name String  @db.VarChar(32)
  age  Int     @db.SmallInt
  coef Decimal @db.Decimal(5, 2)

  @@id([name, age])
}

and here the call 

```
await prisma.corresponding_grams.create({
            data:{
                calories: 10.20,
                plan: "full",
                size: 600.09
            }
        })
```

**Comment by pantharshit00** [comment]
I can reproduce this, thanks for reporting. 

This looks like a duplicate of https://github.com/prisma/prisma/issues/7097 on a first look but data types are different so I will keep this open. 

```
PANIC in libs\prisma-models\src\record.rs:161:30
Invalid coercion encountered: ConversionFailure("Float(BigDecimal(\"10.2\"))", "Decimal")
```

**Comment by tkhatibi** [comment]
I'm using 3.1.1 and ran into it too.

**Comment by tkhatibi** [comment]
It's not a workaround, but panic no longer occurs with [this](https://github.com/prisma/prisma/issues/7061#issuecomment-840487371).

**Comment by butofleury** [comment]
Any news on that? I have the similar issue with sql server. I'm using prisma 3.11.1. Tried also with 3.12.0-dev.43.

**Comment by drico** [comment]
Same for me, and even worse with really big int :

```
 await prisma.myTable.update({
    where: { id: new Prisma.Decimal(45898000) },
    data: {
      comment: 'xxx',
    },
  });
```
with schema : 
```
model TblNewSamp {
  id               Decimal   @id @db.Decimal(10, 0)
  comment   String
}
```

Leads to error : 
```
Error in domain logic: Conversion from `Float(BigDecimal("45898"))` to `Decimal` failed.
    at Object.request (.../node_modules/@prisma/client/runtime/index.js:45409:15)
    at PrismaClient._request (.../node_modules/@prisma/client/runtime/index.js:46301:18)
```

In my case, I see 2 problems : 
- It throws a domain logic error
- The value in the error is not correct : 45898 instead of 45898000) !

**Comment by SevInf** [comment]
Fixed by #2988 and does not reproduce anymore.
